src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionManagerViewModel.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Microsoft.Common.Core.Shell;
+using Microsoft.Common.Wpf;
+using Microsoft.Common.Wpf.Collections;
+using Microsoft.R.Components.ConnectionManager.ViewModel;
+using Microsoft.R.Components.Extensions;
+
+namespace Microsoft.R.Components.ConnectionManager.Implementation.ViewModel {
+    internal sealed class ConnectionManagerViewModel : BindableBase, IConnectionManagerViewModel {
+        private readonly IConnectionManager _connectionManager;
+        private readonly ICoreShell _coreShell;
+        private readonly BatchObservableCollection<object> _items;
+        private IConnectionViewModel _selectedConnection;
+
+        public ConnectionManagerViewModel(IConnectionManager connectionManager, ICoreShell coreShell) {
+            _connectionManager = connectionManager;
+            _coreShell = coreShell;
+            _items = new BatchObservableCollection<object>();
+            Items = new ReadOnlyObservableCollection<object>(_items);
+
+            UpdateConnections();
+        }
+
+        public Task<int> Search(string searchString, CancellationToken cancellationToken) {
+            throw new System.NotImplementedException();
```

Comment:
Search is disabled for now. I'm not really sure if we need it at all - how many connections can one user have?

---
In reply to: [75949265](https://github.com/Microsoft/RTVS/pull/2192#discussion_r75949265) [](ancestors = 75949265)